test/e2e/fixtures/when.go

```diff
@@ -302,6 +302,36 @@ func (w *When) WaitForWorkflow(options ...interface{}) *When {
 	}
 }
 
+func (w *When) WaitForWorkflowList(listOptions metav1.ListOptions, condition func(list []wfv1.Workflow) bool) *When {
+	w.t.Helper()
+	timeout := defaultTimeout
+	start := time.Now()
+	_, _ = fmt.Println("Waiting", timeout.String(), "for workflows", listOptions)
+	timeoutCh := make(chan bool, 1)
```

Comment:
minor - but this is really easy to do using `context.WithTimeout`

Reply:
Will update during the evening after fixing some other things, you are really fast with the feedback, appreciate it!